apps/web/src/components/Onboarding/SetupPassword/SetupPassword.tsx

```diff
@@ -42,10 +44,14 @@ type FormFields = {
 type Mode = "mnemonic" | "secret_key" | "new_mnemonic";
 
 type SetupPasswordProps = {
-  mode: Mode;
+  mode?: Mode;
+  handleProceedToVerification?: (password: string) => void;
 };
 
-export const SetupPassword = ({ mode }: SetupPasswordProps) => {
+export const SetupPassword = ({
+  mode,
+  handleProceedToVerification: handleProceedToVerification,
```

Comment:
so, as I understand, it's just another mode? maybe "verification"? then you don't need to add another, you can just include this bit into the existing `onSubmit` callback

Reply:
also, it's missing lacking tests